modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectFieldFormBase.tsx

```diff
@@ -531,29 +529,92 @@ export function useObjectFieldForm({
 function AggregationSourceProperty({
 	disabled,
 	errors,
-	objectRelationships,
+	editingField,
+	objectDefinitionId,
 	objectFieldSettings = [],
 	setValues,
 }: IAggregationSourcePropertyProps) {
 	const [query, setQuery] = useState<string>('');
-	const [selectedRelatedObject, setSelectRelatedObject] = useState<
-		TObjectRelationship
-	>();
+	const [
+		selectedRelatedObjectDefinition,
+		setSelectRelatedObjectDefinition,
+	] = useState<TObjectRelationship>();
 	const [selectedSummarizeField, setSelectedSummarizeField] = useState<
 		string
 	>();
 	const [
 		selectedAggregationFunction,
 		setSelectedAggregationFunction,
 	] = useState<{label: string; value: string}>();
+	const [objectRelationships, setObjectRelatonships] = useState<
+		TObjectRelationship[]
+	>();
 	const [objectRelationshipFields, setObjectRelationshipFields] = useState<
 		ObjectField[]
 	>();
 
-	const handleChangeRelatedObject = async (
+	useEffect(() => {
```

Comment:
@danvb112 I think this could be a `useMemo`, since the value of `objectDefinitionId` is not being changed inside this component.